lib/chef/provider_resolver.rb

```diff
@@ -68,27 +68,39 @@ def provided_by?(provider_class)
       potential_handlers.include?(provider_class)
     end
 
-    def self.includes_handler?(resource_name, provider_class)
-      priority_map.includes_handler?(resource_name, provider_class)
-    end
-
     def enabled_handlers
-      @enabled_handlers ||=
-        potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
+      potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
```

Comment:
I'm not super attached, like I said; it's just my default to leave things
on the stack wherever possible, and optimize when I must. We certainly
don't want to redo the enabled handlers work, so we should cache that at
least.

On Wed, Jul 1, 2015, 5:25 PM Lamont Granquist <notifications@github.com>
wrote:

> In lib/chef/provider_resolver.rb
> <https://github.com/chef/chef/pull/3627#discussion_r33735422>:
>
> >      def enabled_handlers
> > -      @enabled_handlers ||=
> > -        potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
> > +      potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
>
> Even though this gets called significantly though I doubt its generating
> enough garbage to tax the GC compared to other things that we do, and the
> objects will be short-lived since the provider-resolver object itself is
> short lived. The Ruby 2.x generational GC should handle it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chef/chef/pull/3627/files#r33735422>.
>